src/kboot.c

```diff
@@ -330,13 +332,72 @@ static int dt_set_memory(void)
     printf("FDT: DRAM at 0x%lx size 0x%lx\n", dram_base, dram_size);
     printf("FDT: Usable memory is 0x%lx..0x%lx (0x%lx)\n", dram_min, dram_max, dram_max - dram_min);
 
-    u64 memreg[2] = {cpu_to_fdt64(dram_min), cpu_to_fdt64(dram_max - dram_min)};
+    struct {
+        fdt64_t start;
+        fdt64_t size;
+    } memreg[MAX_MEM_REGIONS] = {{cpu_to_fdt64(dram_min), cpu_to_fdt64(dram_max - dram_min)}};
+    int num_regions = 1;
+
+    int resv_node = fdt_path_offset(dt, "/reserved-memory");
+    if (resv_node < 0)
+        bail("FDT: '/reserved-memory' not found\n");
```

Comment:
I think this should skip over the `fdt_for_each_subnode`. Our production devicetrees will always have `/reserved-memory` but for initial bringup it would be easier without `bail()`